vassal-app/src/main/java/VASSAL/build/module/dice/DieServer.java

```diff
@@ -237,17 +252,17 @@ public void reportResult(RollSet mroll, FormattedString format) {
           total += result;
         }
         else {
-          val += result;
+          val.append(result);
           if (j < nDice - 1)
-            val += ",";
+            val.append(",");
         }
       }
 
       if (reportTotal)
-        val += total;
+        val.append(total);
 
-      val = formatResult(roll.getDescription(), val, format);
-      GameModule.getGameModule().getChatter().send(val);
+      val = new StringBuilder(formatResult(roll.getDescription(), val.toString(), format));
```

Comment:
Why are we making a second `StringBuilder` here?

Reply:
Replaced that and the next line with

   GameModule.getGameModule().getChatter().send(formatResult(roll.getDescription(), val.toString(), format));